### PR TITLE
feat(modules_testdata_base_path): update guidelines for test data

### PR DESCRIPTION
**Pull request by maxulysse**
@netlify /docs/contributing/test_data_guidelines

### CODE REVIEW

File: src/content/docs/contributing/test_data_guidelines.md [hunk 1]
```diff
@@ -57,10 +57,9 @@ In contrast, if your module requires many tests, steps, or heavy CPU usage, you
 
 11. The test data pull request requires a review to be merged.
 
-12. Once you've had your pull request merged in into `nf-core/test-datasets`, make another PR into [nf-core/modules](https://github.com/nf-core/modules), to add your file as a new entry to the [`modules/tests/config/test_data.config`](https://github.com/nf-core/modules/blob/master/tests/config/test_data.config).
+12. Once you've had your pull request merged in into `nf-core/test-datasets`, you are good to go and you can use the newly created file(s) for your test. Use `params.modules_testdata_base_path` to point to the test data directory in your module, and append to it the path to the file(s) you need. This is an example using the `fasta` file from `sarscov2`:
```
Comment (jfy133) [inline src/content/docs/contributing/test_data_guidelines.md:60]:
```suggestion
12. Once you've had your pull request merged in into `nf-core/test-datasets`, you are good to go and you can use refer to the newly uploaded file(s) for your tests. Use `params.modules_testdata_base_path` to point to the test data directory in your module, and append to it the path to the file(s) you need. an example using the `fasta` file from `sarscov2`:
```

File: src/content/docs/contributing/test_data_guidelines.md [hunk 2]
```diff
@@ -57,10 +57,9 @@ In contrast, if your module requires many tests, steps, or heavy CPU usage, you
...
 
-    - This only applies field-specific collections, `delete_me/` files are excluded from this, and raw github URLs should be used in tests.
-    - The 'key' for each URL should follow the style of the full file name with extensions but with underscores rather than full-stops e.g. `genome.fa.gz` would become `genome_fa_gz`.
+    - `file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)`
```
Comment (jfy133) [inline src/content/docs/contributing/test_data_guidelines.md:62]:
```suggestion
    - `file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)`
    - Note the file string corresponds to the directory structure of the modules repository.
```

Comment (jfy133) [reply src/content/docs/contributing/test_data_guidelines.md:62]: To be completely explicit (as the old map system didn't exactly map that always)

File: src/content/docs/contributing/test_data_guidelines.md [hunk 3]
```diff
@@ -57,10 +57,10 @@ In contrast, if your module requires many tests, steps, or heavy CPU usage, you
 
 11. The test data pull request requires a review to be merged.
 
-12. Once you've had your pull request merged in into `nf-core/test-datasets`, make another PR into [nf-core/modules](https://github.com/nf-core/modules), to add your file as a new entry to the [`modules/tests/config/test_data.config`](https://github.com/nf-core/modules/blob/master/tests/config/test_data.config).
+12. Once you've had your pull request merged in into `nf-core/test-datasets`, you are good to go and you can use refer to the newly uploaded file(s) for your tests. Use `params.modules_testdata_base_path` to point to the test data directory in your module, and append to it the path to the file(s) you need. an example using the `fasta` file from `sarscov2`:
```
Comment (maxulysse) [inline src/content/docs/contributing/test_data_guidelines.md:60]:
```suggestion
12. Once you've had your pull request merged in into `nf-core/test-datasets`, you are good to go and you can use refer to the newly uploaded file(s) for your test(s). Use `params.modules_testdata_base_path` to point to the test data directory in your module, and append to it the path to the file(s) you need. an example using the `fasta` file from `sarscov2`:
```

File: src/content/docs/contributing/test_data_guidelines.md [hunk 4]
```diff
@@ -57,10 +57,10 @@ In contrast, if your module requires many tests, steps, or heavy CPU usage, you
 
 11. The test data pull request requires a review to be merged.
 
-12. Once you've had your pull request merged in into `nf-core/test-datasets`, make another PR into [nf-core/modules](https://github.com/nf-core/modules), to add your file as a new entry to the [`modules/tests/config/test_data.config`](https://github.com/nf-core/modules/blob/master/tests/config/test_data.config).
+12. Once you've had your pull request merged in into `nf-core/test-datasets`, you are good to go and you can use refer to the newly uploaded file(s) for your test(s). Use `params.modules_testdata_base_path` to point to the test data directory in your module, and append to it the path to the file(s) you need. an example using the `fasta` file from `sarscov2`:
```
Comment (mashehu) [inline src/content/docs/contributing/test_data_guidelines.md:60]:
```suggestion
12. Once you've had your pull request merged in into `nf-core/test-datasets`, you are good to go and you can refer to the newly uploaded file(s) for your test(s). Use `params.modules_testdata_base_path` to point to the test data directory in your module, and append to it the path to the file(s) you need. an example using the `fasta` file from `sarscov2`:
```

File: src/content/docs/contributing/test_data_guidelines.md [hunk 5]
```diff
@@ -57,10 +57,10 @@ In contrast, if your module requires many tests, steps, or heavy CPU usage, you
...
 
-    - This only applies field-specific collections, `delete_me/` files are excluded from this, and raw github URLs should be used in tests.
-    - The 'key' for each URL should follow the style of the full file name with extensions but with underscores rather than full-stops e.g. `genome.fa.gz` would become `genome_fa_gz`.
+    - `file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)`
```
Comment (mashehu) [inline src/content/docs/contributing/test_data_guidelines.md:62]:
````suggestion 
```groovy
    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
 ```
````

File: src/content/docs/contributing/test_data_guidelines.md [hunk 6]
```diff
@@ -57,10 +57,10 @@ In contrast, if your module requires many tests, steps, or heavy CPU usage, you
...
+    - Note the file string corresponds to the directory structure of the modules repository.
```
Comment (mashehu) [inline src/content/docs/contributing/test_data_guidelines.md:63]:
```suggestion
 Note that the file string corresponds to the directory structure of the modules repository.
```

**Review by mashehu — APPROVED** [review]

**Comment by mashehu** [comment]
@maxulysse what is the status of this?

File: sites/docs/src/content/docs/tutorials/tests_and_test_data/test_data.md [hunk 7]
```diff
@@ -58,6 +58,14 @@ In contrast, if your module requires many tests, steps, or heavy CPU usage, you
 
 11. The test data pull request requires a review to be merged.
 
+12. Once you've had your pull request merged in into `nf-core/test-datasets`, you are good to go and you can refer to the newly uploaded file(s) for your test(s). Use `params.modules_testdata_base_path` to point to the test data directory in your module, and append to it the path to the file(s) you need. Here is an example using the `fasta` file from `sarscov2`:
```
Comment (jfy133) [inline sites/docs/src/content/docs/tutorials/tests_and_test_data/test_data.md:61]:
```suggestion
12. Once you've had your pull request merged in into `nf-core/test-datasets`, you are good to go and you can refer to the newly uploaded file(s) for your test(s). Use `params.modules_testdata_base_path` to point to the test data directory in your module, and append to it the path to the file(s) you need. Here is an example using the `fasta` file from `sarscov2`:
```

I don't get waht you mean by 'to point to the test data directory in your module'

It points to the remote URL (on GitHbub) or local source ( a clone of teh repo) of all the  testr data

Can you rephrase?

File: sites/docs/src/content/docs/tutorials/tests_and_test_data/test_data.md [hunk 8]
```diff
@@ -58,6 +58,14 @@ In contrast, if your module requires many tests, steps, or heavy CPU usage, you
 
 11. The test data pull request requires a review to be merged.
 
+12. Once you've had your pull request merged in into `nf-core/test-datasets`, you are good to go and you can refer to the newly uploaded file(s) for your test(s). `params.modules_testdata_base_path` points to `https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/` in the modules repo, and refers to the `data/` folder in the `modules` branch. Append to it the path to the file(s) you need. Here is an example using the `fasta` file from `sarscov2`:
```
Comment (maxulysse) [inline sites/docs/src/content/docs/tutorials/tests_and_test_data/test_data.md:61]:
```suggestion
12. Once you've had your pull request merged in into `nf-core/test-datasets`, you are good to go and you can refer to the newly uploaded file(s) for your test(s). `params.modules_testdata_base_path` points to `https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/` from the [test-datasets](https://github.com/nf-core/test-datasets) repo, and refers to the `data/` folder in the `modules` branch. Append to it the path to the file(s) you need. Here is an example using the `fasta` file from `sarscov2`:
```

File: sites/docs/src/content/docs/tutorials/tests_and_test_data/test_data.md [hunk 9]
```diff
@@ -58,6 +58,14 @@ In contrast, if your module requires many tests, steps, or heavy CPU usage, you
 
 11. The test data pull request requires a review to be merged.
 
+12. Once you've had your pull request merged in into `nf-core/test-datasets`, you are good to go and you can refer to the newly uploaded file(s) for your test(s). `params.modules_testdata_base_path` points to `https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/` from the [test-datasets](https://github.com/nf-core/test-datasets) repo, and refers to the `data/` folder in the `modules` branch. Append to it the path to the file(s) you need. Here is an example using the `fasta` file from `sarscov2`:
```
Comment (jfy133) [inline sites/docs/src/content/docs/tutorials/tests_and_test_data/test_data.md:61]:
```suggestion
12. Once you've had your pull request merged in into `nf-core/test-datasets`, you are good to go and you can refer to the newly uploaded file(s) for your test(s). By default `params.modules_testdata_base_path` points to `https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/` from the [test-datasets](https://github.com/nf-core/test-datasets) repo, and refers to the `data/` folder in the `modules` branch. Append to it the path to the file(s) you need. Here is an example using the `fasta` file from `sarscov2`:
```

**Review by jfy133 — APPROVED** [review]
